controller/backup_controller.go

```diff
@@ -248,17 +248,18 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 
 	log := getLoggerForBackup(bc.logger, backup)
 
-	// Get default backup target
-	backupTarget, err := bc.ds.GetBackupTargetRO(types.DefaultBackupTargetName)
+	backupTarget, err := bc.ds.GetBackupTargetRO(backup.Spec.BackupTargetName)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			return nil
+		if !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed to get the backup target %v", backup.Spec.BackupTargetName)
+		}
+		if backup.DeletionTimestamp == nil {
```

Comment:
Suggest change to as comment below in `backup_volume_controller.go`
```
	backupTarget, err := bc.ds.GetBackupTargetRO(...)
	if err != nil && !apierrors.IsNotFound(err) {
		return errors.Wrapf(err, "failed to get backup target")
	}

        if backupTarget == nil && CR.DeletionTimestamp == nil {
        	return fmt.Errorf("Failed to find the backup target for the CR...")
        }
```

Reply:
Modified. And I think I might try `ownerReferences` to have a dependent deletion.

Reply:
>Will it become an orphan CR?

Yes, it might be an orphan CR.

>When will this situation happen?

I believe that it will occur if the first backup is created and the backup volume is not created yet (because the backup is not completed), then we delete the backup target.